main/src/ca/uwaterloo/flix/language/phase/Lexer.scala

```diff
@@ -59,8 +59,14 @@ object Lexer {
     '&' -> TokenKind.Ampersand,
     '|' -> TokenKind.Bar,
     '^' -> TokenKind.Caret,
+    '$' -> TokenKind.Dollar,
   )
 
+  /**
+   * Since Flix support hex decimals, a 'digit' can also be some select characters.
+   */
+  private val DIGITS = List('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'a', 'b', 'c', 'd', 'e', 'f')
```

Comment:
Better to define this as a function and compare if the char is between two numbers.